sql/core/src/main/scala/org/apache/spark/sql/execution/Queryable.scala

```diff
@@ -27,6 +28,7 @@ private[sql] trait Queryable {
   def schema: StructType
   def queryExecution: QueryExecution
   def sqlContext: SQLContext
+  private[sql] def logicalPlan: LogicalPlan
```

Comment:
Can we just get this from the `queryExecution` in the cache manager?  or at least define it explicitly here.  I don't want dataframes and datasets to fall out of sync with regards to what the canonical plan phase is.